source/core/interface_tools.cc

```diff
@@ -0,0 +1,299 @@
+// SPDX-FileCopyrightText: Copyright (c) 2025 The Lethe Authors
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception OR LGPL-2.1-or-later
+
+#include <core/interface_tools.h>
+
+
+template <int dim, typename VectorType, typename FEType>
+InterfaceTools::CellWiseFunction<dim, VectorType, FEType>::CellWiseFunction(
+  const unsigned int p_fe_degree)
+  : fe(p_fe_degree)
+{
+  n_cell_wise_dofs = fe.dofs_per_cell;
+}
+
+template <int dim, typename VectorType, typename FEType>
+void
+InterfaceTools::CellWiseFunction<dim, VectorType, FEType>::set_active_cell(
+  const VectorType &in_local_dof_values)
+{
+  cell_dof_values = in_local_dof_values;
+}
+
+template <int dim, typename VectorType, typename FEType>
+double
+InterfaceTools::CellWiseFunction<dim, VectorType, FEType>::value(
+  const Point<dim>  &point,
+  const unsigned int component) const
+{
+  double value = 0;
+  for (unsigned int i = 0; i < n_cell_wise_dofs; ++i)
+    value += cell_dof_values[i] * fe.shape_value_component(i, point, component);
+
+  return value;
+}
+
+template <int dim, typename VectorType, typename FEType>
+Tensor<1, dim>
+InterfaceTools::CellWiseFunction<dim, VectorType, FEType>::gradient(
+  const Point<dim>  &point,
+  const unsigned int component) const
+{
+  Tensor<1, dim> gradient;
+  for (unsigned int i = 0; i < n_cell_wise_dofs; ++i)
+    gradient +=
+      cell_dof_values[i] * fe.shape_grad_component(i, point, component);
+
+  return gradient;
+}
+
+template <int dim, typename VectorType, typename FEType>
+SymmetricTensor<2, dim>
+InterfaceTools::CellWiseFunction<dim, VectorType, FEType>::hessian(
+  const Point<dim>  &point,
+  const unsigned int component) const
+{
+  Tensor<2, dim> hessian;
+  for (unsigned int i = 0; i < n_cell_wise_dofs; ++i)
+    hessian +=
+      cell_dof_values[i] * fe.shape_grad_grad_component(i, point, component);
+
+  return symmetrize(hessian);
+}
```

Comment:
These functions are small and tiny, why not inline them instead in the .h directly?